pennylane_lightning/src/simulators/lightning_kokkos/gates/GateFunctorsParam.hpp

```diff
@@ -207,39 +202,38 @@ template <class PrecisionT, bool inverse = false> struct twoQubitOpFunctor {
 
 template <class PrecisionT, bool inverse = false> struct multiQubitOpFunctor {
 
-    using KokkosComplexVector = Kokkos::View<Kokkos::complex<PrecisionT> *>;
-    using KokkosIntVector = Kokkos::View<std::size_t *>;
+    using ComplexT = Kokkos::complex<PrecisionT>;
+    using KokkosComplexVector = Kokkos::View<ComplexT *>;
+    using KokkosIntVector = Kokkos::View<size_t *>;
 
     KokkosComplexVector arr;
     KokkosComplexVector matrix;
-    KokkosIntVector indices;
     KokkosIntVector wires;
-    KokkosComplexVector coeffs_in;
-    std::size_t dim;
-    std::size_t num_qubits;
+    size_t dim;
+    size_t num_qubits;
 
-    multiQubitOpFunctor(KokkosComplexVector &arr_, std::size_t num_qubits_,
+    multiQubitOpFunctor(KokkosComplexVector &arr_, size_t num_qubits_,
```

Comment:
Sorry @mlxd @AmintorDusko , the meaningful changes are from here down to line 229.

Reply:
In essence, we need to define `indices` and and `coeff_in` as local variables.

Reply:
Gotcha, thanks!

And the previously disabled test that was failing has been re-enabled and is now passing?

Reply:
Yes, this is the modified test file. Random unitary gates are applied to a statevector. Previously, only 1- and 2-qubit gates were tested, but these have their own special functor, so `multiQubitOpFunctor` was not hit. We now go up to 3- and 4-qubit gates, which does hit `multiQubitOpFunctor`.

Reply:
Thank you @vincentmr, for guiding our review efforts.